File: src/main/chungmu/src/js/SearchResult.js
class SearchResult {
    constructor(title, creator, imageLink, videoLink) {
        this.title = title;
        this.creator = creator;
        this.imageLink = imageLink;
        this.videoLink = videoLink;
    }

    // Getter와 Setter 메서드
    getTitle() {
        return this.title;
    }

    setTitle(title) {
        this.title = title;
    }

    getCreator() {
        return this.creator;
    }

    setCreator(creator) {
        this.creator = creator;
    }

    getImageLink() {
        return this.imageLink;
    }

    setImageLink(imageLink) {
        this.imageLink = imageLink;
    }

    getVideoLink() {
        return this.videoLink;
    }

    setVideoLink(videoLink) {
        this.videoLink = videoLink;
    }

    // toString 메서드 (선택사항)
    toString() {
        return `Track { title: '${this.title}', creator: '${this.creator}', imageLink: '${this.imageLink}', videoLink: '${this.videoLink}' }`;
    }
}

File: src/main/java/com/sophra/chungmusic/SearchResult.java
package com.sophra.chungmusic;

public class SearchResult {
    private String title;
    private String creator;
    private String imageLink;
    private String videoLink;

    // 기본 생성자
    public SearchResult() {
    }

    public SearchResult(String title, String creator, String imageLink, String videoLink) {
        this.title = title;
        this.creator = creator;
        this.imageLink = imageLink;
        this.videoLink = videoLink;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getCreator() {
        return creator;
    }

    public void setCreator(String creator) {
        this.creator = creator;
    }

    public String getImageLink() {
        return imageLink;
    }

    public void setImageLink(String imageLink) {
        this.imageLink = imageLink;
    }

    public String getVideoLink() {
        return videoLink;
    }

    public void setVideoLink(String videoLink) {
        this.videoLink = videoLink;
    }

    // 오버라이딩된 toString() 메서드 (선택사항)
    @Override
    public String toString() {
        return "Track{" +
                "title='" + title + '\'' +
                ", creator='" + creator + '\'' +
                ", imageLink='" + imageLink + '\'' +
                ", videoLink='" + videoLink + '\'' +
                '}';
    }
}
